tensorflow/lite/micro/kernels/reduce_common.cc

```diff
@@ -111,11 +117,23 @@ TfLiteStatus PrepareMeanOrSumHelper(TfLiteContext* context, TfLiteNode* node,
     op_data->output_scale = output->params.scale;
   }
 
+  context->RequestScratchBufferInArena(context, output_size * sizeof(int32_t),
+                                       &op_data->temp_index_idx);
+  context->RequestScratchBufferInArena(context, output_size * sizeof(int32_t),
+                                       &op_data->temp_buffer_idx);
```

Comment:
Maybe I am misunderstanding something.  input can be either type int8 or int16,  but output is fixed to be int32_t?  Maybe worthy of documenting the reason of using sizeof(int32_t) here

Reply:
Since the change to the reference implementation of reduce is going to be rolled-back in the near future, let's merge this as is.